lightning/src/routing/scoring.rs

```diff
@@ -2870,4 +2871,46 @@ mod tests {
 		};
 		assert_eq!(scorer.channel_penalty_msat(42, &source, &target, usage), 0);
 	}
+
+	#[test]
+	fn scores_with_blinded_path() {
+		// Make sure we'll account for a blinded path's final_value_msat in scoring
+		let logger = TestLogger::new();
+		let network_graph = network_graph(&logger);
+		let params = ProbabilisticScoringParameters {
+			liquidity_penalty_multiplier_msat: 1_000,
+			liquidity_offset_half_life: Duration::from_secs(10),
+			..ProbabilisticScoringParameters::zero_penalty()
+		};
+		let mut scorer = ProbabilisticScorer::new(params, &network_graph, &logger);
+		let source = source_node_id();
+		let target = target_node_id();
+		let usage = ChannelUsage {
+			amount_msat: 512,
+			inflight_htlc_msat: 0,
+			effective_capacity: EffectiveCapacity::Total { capacity_msat: 1_024, htlc_maximum_msat: 1_000 },
+		};
+		assert_eq!(scorer.channel_penalty_msat(42, &source, &target, usage), 300);
+
+		let mut path = payment_path_for_amount(768);
+		let recipient_hop = path.hops.pop().unwrap();
+		let blinded_path = BlindedPath {
+			introduction_node_id: path.hops.last().as_ref().unwrap().pubkey,
+			blinding_point: test_utils::pubkey(42),
+			blinded_hops: vec![
+				BlindedHop { blinded_node_id: test_utils::pubkey(44), encrypted_payload: Vec::new() }
+			],
+		};
+		path.blinded_tail = Some(BlindedTail {
+			hops: blinded_path.blinded_hops,
+			blinding_point: blinded_path.blinding_point,
+			final_cltv_expiry_delta: recipient_hop.cltv_expiry_delta,
+			final_value_msat: recipient_hop.fee_msat,
+		});
+
+		scorer.payment_path_failed(&path, 42);
+		path.blinded_tail.as_mut().unwrap().final_value_msat = 256;
+		scorer.payment_path_failed(&path, 43);
+		assert_eq!(scorer.channel_penalty_msat(42, &source, &target, usage), 281);
```

Comment:
Not sure I understand why `final_value_msat` is changed and why the penalty is only checked for `42` and not `43`.

Reply:
These checks/numbers are taken from `restricts_liquidity_bounds_after_decay`. Added a comment and checked `43`

Reply:
But why is `final_value_msat` being changed?

Reply:
I may be misunderstanding, I thought that improved the confidence interval of the scorer, leading to a lower penalty. I mostly just wanted to use numbers from a test that's already known to be accurate.

Reply:
Hmm.. I don't see that being set in `restricts_liquidity_bounds_after_decay`? We should use a realistic example instead of changing the value, unless there is a good reason to do otherwise (e.g., simulating something that couldn't otherwise be done). Having the final value change between scoring each hop isn't a realistic scenario.

Reply:
Discussed offline. I confused myself on which channels were being scored on each call to `payment_path_failed`. But I think we should just check that the liquidity balances were correctly changed rather than caring about what scores are being returned, like in `resets_liquidity_upper_bound_when_crossed_by_lower_bound`.

Reply:
Switched to check liquidity balances instead